app/frontend/src/service/conversations.ts

```diff
@@ -125,3 +127,18 @@ export function markLastSeenGroupChat(
   req.setLastSeenMessageId(lastSeenMessageId);
   return client.conversations.markLastSeenGroupChat(req);
 }
+
+export async function getDirectMessage(userId: number) {
+  const req = new GetDirectMessageReq();
+  req.setUserId(userId);
+  try {
+    const res = await client.conversations.getDirectMessage(req);
+    return res.getGroupChatId();
+  } catch (e) {
+    if (e?.code === StatusCode.NOT_FOUND) {
```

Comment:
Probably more of a backend API nitpick, but still feels a bit shady to me to throw an error on not found when it's not something the user could have done to cause this error... it might be cleaner if we can return some extra boolean like `hasExistingDm` or something that we can check on in the frontend and put the response under `oneof`? I think that's the proto equivalent of a union type right even though the generated code doesn't really do it?

Reply:
Not saying you have to do it in this here though, maybe something to think about in the upcoming request/message API refactor?